spec/controllers/admin/membership_submissions_controller_spec.rb

```diff
@@ -9,7 +9,89 @@
     end
   end
 
-  pending "GET #pending"
-  pending "POST #pending"
-  pending "POST #reject"
+  describe "GET #pending" do
+    describe "when pending submissions are zero" do
+      before do
+        allow(@controller).to receive(:require_admin!).and_return(true)
+        get :pending
+      end
+
+      it "redirects to admin_path" do
+        expect(response).to redirect_to(admin_path)
+      end
+
+      it "sets flash/info" do
+        expect(flash[:info]).to match(/submissions/i)
+      end
+    end
+
+    describe "when pending submissions are grater than zero" do
+      before do
+        @slack_membership_1 = create(:slack_membership_submission, email: 'johndoe@email.com')
+        @slack_membership_2 = create(:slack_membership_submission, email: 'johndoe2@email.com')
+        allow(@controller).to receive(:require_admin!).and_return(true)
+        get :pending
+      end
+
+      it "does not set flash/info and returns first membership submission" do
+        expect(flash[:info]).to be_blank
+        expect(SlackMembershipSubmission.find_by_email('johndoe@email.com')).to eq @slack_membership_1
+      end
+    end
+  end
+
+  describe "GET #approve" do
+    describe "when pending submissions are zero" do
+      before do
+        @slack_membership = create(:slack_membership_submission, email: 'johndoe@email.com', status: 1)
+        allow(@controller).to receive(:require_admin!).and_return(true)
+        get :approve, {params: {id: @slack_membership.id}}
+      end
+
+      it "does set flash/info" do
+        expect(flash[:warning]).to match(/approved/i)
```

Comment:
Again, a bit more context for the expected match would be good.  The above would also match "the submission was not approved" which is different from what we want.